src/app/tasks/task-details/task-details.component.ts

```diff
@@ -49,50 +50,67 @@ export class TaskDetailsComponent implements OnInit, OnDestroy {
   onSubmit () {
     // Get filled out form data using form group
 
-    this.taskService.updateTask(
-      this.task.id,
-      {
-        id: this.task.id,
-        title: this.editTaskForm.get('title')?.value,
-        description: this.editTaskForm.get('description')?.value,
-        type: this.taskTypes.find( taskType => 
-          taskType.value == this.editTaskForm.get('type')?.value
-        ) as Option,
-        status: this.taskStatuses.find(taskStatus => 
-          taskStatus.value == this.editTaskForm.get('status')?.value
-        ) as Option,
-        modifiedOn: new Date(),
-        createdOn: this.task.createdOn,
-      }
-    );
-
-    this.router.navigate(['tasks']);
+    if(this.task) {
+      this.taskService.updateTask(
+        this.task.id,
+        {
+          id: this.task.id,
+          title: this.editTaskForm.get('title')?.value,
+          description: this.editTaskForm.get('description')?.value,
+          type: this.taskTypes.find( taskType => 
+            taskType.value == this.editTaskForm.get('type')?.value
+          ) as Option,
+          status: this.taskStatuses.find(taskStatus => 
+            taskStatus.value == this.editTaskForm.get('status')?.value
+          ) as Option,
+          updatedAt: new Date(),
+          createdAt: this.task.createdAt,
+        }
+      ).subscribe( updatedTask => {
+        if(updatedTask) {
+          this.router.navigateByUrl('/', { skipLocationChange: true }).then(() => {
+            this.router.navigate(['/tasks', this.task?.id])});
+        }
+        else this.router.navigate(['/tasks']);
+      });
+  
+    }    
   }
 
   constructor(private router: Router, private route: ActivatedRoute, private taskService: TaskService) {}
+  ngOnDestroy(): void {
+    this.deleteTaskSubscription.unsubscribe();
+    this.getCurrentTaskSubscription.unsubscribe();
```

Comment:
One pointer: If you only need to make a request once (like fetching task data initally) you can use the take(1) rxjs operator to take only 1 value from the subscribe, so it woud look something like this this.getCurrentTask().pipe(take(1)).subscribe(res => etc);